cloud_governance/common/helpers/aws/aws_cleanup_operations.py

```diff
@@ -0,0 +1,153 @@
+import datetime
+
+import boto3
+
+from cloud_governance.common.clouds.aws.s3.s3_operations import S3Operations
+from cloud_governance.common.helpers.cleanup_operations import AbstractCleanUpOperations
+from cloud_governance.common.logger.init_logger import logger
+
+
+class AWSCleanUpOperations(AbstractCleanUpOperations):
+
+    def __init__(self):
+        super().__init__()
+        self._region = self._environment_variables_dict.get('AWS_DEFAULT_REGION', 'us-east-2')
+        self.__s3operations = S3Operations(region_name=self._region)
+        self._ec2_client = boto3.client('ec2', region_name=self._region)
+        self._s3_client = boto3.client('s3')
+        self._iam_client = boto3.client('iam')
+
+    def get_tag_name_from_tags(self, tags: list, tag_name: str) -> str:
+        if tags:
+            for tag in tags:
+                if tag.get('Key').strip().lower() == tag_name.lower():
+                    return tag.get('Value').strip()
+        return ''
+
+    def get_clean_up_days_count(self, tags: list):
+        """
+        This method returns the cleanup days count
+        :param tags:
+        :type tags:
+        :return:
+        :rtype:
+        """
+        if self._dry_run == 'no':
+            last_used_day = self.get_tag_name_from_tags(tags=tags, tag_name='DryRunNoDays')
+        else:
+            last_used_day = self.get_tag_name_from_tags(tags=tags, tag_name='DryRunYesDays')
+        if not last_used_day:
+            return 1
+        else:
+            date, days = last_used_day.split('@')
+            if date != str(datetime.datetime.now().date()):
+                return int(days) + 1
+            return 1 if int(days) == 0 else int(days)
+
+    def get_skip_policy_value(self, tags: list) -> str:
+        """
+        This method returns the skip value
+        :param tags:
+        :type tags:
+        :return:
+        :rtype:
+        """
+        policy_value = self.get_tag_name_from_tags(tags=tags, tag_name='Policy').strip()
+        if not policy_value:
+            policy_value = self.get_tag_name_from_tags(tags=tags, tag_name='Skip').strip()
+        if policy_value:
+            return policy_value.replace('_', '').replace('-', '').upper()
+        return 'NA'
+
+    def _delete_resource(self, resource_id: str):
+        """
+        This method deletes the resource by verifying the policy
+        :param resource_id:
+        :type resource_id:
+        :return:
+        :rtype:
+        """
+        action = "deleted"
+        try:
```

Comment:
Should replace to case

Reply:
I wonder, does python have the switch case? 🤔